rfcs/1478-variables.md

```diff
@@ -0,0 +1,123 @@
+
+On the server-side, life would be made easier if Seq could offer a similar experience:
+
+ * Save the id of an interesting HTTP request to e.g. `$suspiciousRequestId`, and then easily return to searching for `RequestId = $suspiciousRequestId`
+ * Save a fragment of text like a problematic machine name to e.g. `$slowServer`, and then refer to it multiple times within a query `select count(*) from stream where contains(@Message, $slowServer) or contains(@Exception, $slowServer)`
+ * Assign a large/complex property value to a variable, e.g. `$failingRequestBody`, and then interactively examine elements of it with `select $failingRequestBody.items[?].tag = 'legacy'`
+
+A new _Variables_ tool window can list current variable names and their values, avoiding at least some of the scribbling, copying, pasting, and double-handling that I'd otherwise need to do.
+
+## Detailed design
+
+### Variable names
+
+A variable is a named value. In the Seq query language, variable names start with `$`, so that there's no risk of getting incorrect results when the name of a variable conflicts with the name of an event property or column (neither of these things can use names that start with `$`).
+
+Formally, as a regular expression, variables names match the pattern `\$[A-Za-z_0-9]+`.
+
+Examples of variable names:
+
+ * `$a`
+ * `$TEST`
+ * `$_`
+ * `$1`
+ * `$event99`
+
+### Variable values
+
+Variables can hold any Seq runtime value. For practical purposes, this means JSON-style `null`, Booleans, numbers, strings, arrays, and objects can all be stored in a variable.
+
+### Variable scope
+
+Variables are defined and used within a single Seq browser window. Variables are not persisted or tracked by the Seq server.
+
+The _Variables_ tool window (see below) provides a simple "copy" function that can be used to replicate a variable from one Seq browser session into another.
+
+### Setting, editing, and clearing variables
+
+#### In the _Variables_ tool window
+
+In the right-hand (signal bar) panel, a new tool window called _Variables_ shows the list of variables, and their values, held in the current browser session.
+
+> Some reorganization of the signal bar is planned so that tool windows can be supported.
+
+The _Variables_ tool window has a "+" button, allowing a new variable to be added, alongside features to edit, remove, and clear current variables.
+
+A user of the Seq interface can click the add button, give the variable a name (`$` prefix is enforced), and choose to give the variable a literal text or numeric value.
+
+Variables in the tool window will provide a context menu with _Copy name_, _Copy value_, and _Copy assignment_, where the assignment is a query language statement that sets the variable value.
+
+#### Direcly, in the query language
+
+Variables can be set in the Seq search box using `select into`:
+
+```sql
+select ToIsoString(now()) into $rightNow
```

Comment:
A simpler example would be better, since you are not demonstrating `ToIsoString` or `now`. 

`select 7 into $myNumber`
`select 'Brisbane' into $city`